components/ui/project-list.tsx

```diff
@@ -1,22 +1,40 @@
+"use client";
```

Comment:
Why was "use client" introduced? Is it because of the useSearchStore?

An alternative may be to keep the useSearchStore outside and have this component take a loading prop. Keep it as "projects.length === 0" shows none found. Then set loading in whatever is calling it -- I see there was a bit of a hack put in for category-project-list to make this work